aptos-move/block-executor/src/executor.rs

```diff
@@ -409,13 +413,109 @@ where
         &self,
         txn_idx: TxnIndex,
         versioned_cache: &MVHashMap<T::Key, T::Tag, T::Value, X, T::Identifier>,
+        scheduler: &Scheduler,
+        shared_counter: &AtomicU32,
         last_input_output: &TxnLastInputOutput<T, E::Output, E::Error>,
         base_view: &S,
     ) {
-        let delta_keys = last_input_output.delta_keys(txn_idx);
-        let _events = last_input_output.events(txn_idx);
-        let mut delta_writes = Vec::with_capacity(delta_keys.len());
-        for k in delta_keys.into_iter() {
+        println!("worker commit hook called");
+        let parallel_state = ParallelState::<T, X>::new(versioned_cache, scheduler, shared_counter);
+        let latest_view = LatestView::new(base_view, ViewState::Sync(parallel_state), txn_idx);
+
+        // For each aggregator v2 in resource write set, replace the identifiers with values.
+        let mut write_set_keys = vec![];
+        let resource_write_set = last_input_output.resource_write_set(txn_idx);
+        let mut patched_resource_write_set = HashMap::new();
+        if let Some(resource_write_set) = resource_write_set {
+            for (key, (write_op, layout)) in resource_write_set.iter() {
+                // layout is Some(_) if it contains an aggregator
+                if layout.is_some() && !write_op.is_deletion() {
```

Comment:
Cane use `if let` here to avoid unwraps later

Reply:
if let doesn't yet allow && something else, and adds more nesting into this nested loop. 

I am fine either way